cmake/setup_compiler_flags_gnu.cmake

```diff
@@ -97,11 +97,18 @@ IF(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-variadic-macros")
   ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-c++11-extensions")
 
-  IF(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "3.6")
-    #
-    # Clang versions prior to 3.6 emit a lot of false positives wrt
-    # "-Wunused-function".
-    #
+  #
+  # Clang versions prior to 3.6 emit a lot of false positives wrt
+  # "-Wunused-function". Also suppress warnings for Xcode older than 6.3
+  # (which is equivalent to clang < 3.6).
+  #
+  # FIXME: I wait for the day with a clang version "4.0"... and I will
+  # curse the person that thought it is a _great_ idea to come up with
+  # independent version numbers for clang on Mac...
+  #
+  IF( CMAKE_CXX_COMPILER_VERSION VERSION_LESS "3.6" OR
+      ( NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.0" AND
+        CMAKE_CXX_COMPILER_VERSION VERSION_LESS "6.3") )
```

Comment:
> So what does `CMAKE_CXX_COMPILER_VERSION` actually indicate -- the
> clang version, or the Xcode version?

Both. There is no difference in CMAKE_CXX_COMPILER_ID for both
compilers.

> The code currently triggers for
> all versions so that either `VERSION<3.6` or `4.0<=VERSION<6.3`.

Correct. This is the intended behavior.